pkg/kubelet/images/image_manager.go

```diff
@@ -143,19 +171,95 @@ func (m *imageManager) EnsureImageExists(ctx context.Context, objRef *v1.ObjectR
 		RuntimeHandler: podRuntimeHandler,
 	}
 
-	imageRef, message, err = m.imagePullPrecheck(ctx, objRef, logPrefix, pullPolicy, &spec, imgRef)
+	imageRef, message, err = m.imagePullPrecheck(ctx, objRef, logPrefix, pullPolicy, &spec, containerImage)
 	if err != nil {
 		return "", message, err
 	}
+
+	repoToPull, _, _, err := parsers.ParseImageName(spec.Image)
+	if err != nil {
+		return "", err.Error(), err
+	}
+
+	keyring, err := credentialprovidersecrets.MakeDockerKeyring(pullSecrets, m.nodeKeyring)
+	if err != nil {
+		return "", err.Error(), err
+	}
+
+	pullCredentials, _ := keyring.Lookup(repoToPull)
+
 	if imageRef != "" {
```

Comment:
needs a comment.. precheck error'd out for never pull above.. falling through to pull image at bottom of fund

Reply:
I moved the `PullPolicy == Never` check outside the above `imageRef != ""` block and added an explanatory comment.